diff -Naur cvsps-2.1.orig/cbtcommon/text_util.c cvsps-2.1/cbtcommon/text_util.c
--- cvsps-2.1.orig/cbtcommon/text_util.c	2005-05-26 03:39:40.000000000 +0000
+++ cvsps-2.1/cbtcommon/text_util.c	2018-12-04 05:14:44.112405139 +0000
@@ -236,9 +236,6 @@
     }
 }
 
-#ifdef linux
-extern void *memfrob(void *, size_t);
-#else
 static void * memfrob(void * mem, size_t len)
 {
     size_t i;
@@ -252,7 +249,6 @@
 
     return mem;
 }
-#endif
 
 // simple functions to obfuscate strings in a binary
 char* frobstr( char* src )
