diff -Naur dhewm3-1.4.0.orig/neo/idlib/Heap.cpp dhewm3-1.4.0/neo/idlib/Heap.cpp
--- dhewm3-1.4.0.orig/neo/idlib/Heap.cpp	2015-10-09 14:06:49.000000000 +0000
+++ dhewm3-1.4.0/neo/idlib/Heap.cpp	2017-01-08 19:16:43.853342484 +0000
@@ -32,7 +32,7 @@
 #include "idlib/Heap.h"
 
 #ifndef USE_LIBC_MALLOC
-	#define USE_LIBC_MALLOC		0
+	#define USE_LIBC_MALLOC		1
 #endif
 
 #ifndef CRASH_ON_STATIC_ALLOCATION
