diff -Naur ptlib-2.10.11.orig/src/ptclib/pssl.cxx ptlib-2.10.11/src/ptclib/pssl.cxx
--- ptlib-2.10.11.orig/src/ptclib/pssl.cxx	2013-08-14 23:20:27.000000000 +0000
+++ ptlib-2.10.11/src/ptclib/pssl.cxx	2016-12-14 02:05:38.395168637 +0000
@@ -805,15 +805,11 @@
   SSL_METHOD * meth;
 
   switch (method) {
-    case SSLv3:
-      meth = SSLv3_method();
-      break;
     case TLSv1:
       meth = TLSv1_method(); 
       break;
-    case SSLv23:
     default:
-      meth = SSLv23_method();
+      meth = TLSv1_method(); 
       break;
   }
 
