diff -Naur lighttpd-1.4.45.orig/src/network.c lighttpd-1.4.45/src/network.c
--- lighttpd-1.4.45.orig/src/network.c	2017-01-14 05:05:17.000000000 +0000
+++ lighttpd-1.4.45/src/network.c	2017-01-25 09:38:00.824932481 +0000
@@ -841,23 +841,23 @@
 		SSL_CTX_set_options(s->ssl_ctx, ssloptions);
 		SSL_CTX_set_info_callback(s->ssl_ctx, ssl_info_callback);
 
-		if (!s->ssl_use_sslv2 && 0 != SSL_OP_NO_SSLv2) {
-			/* disable SSLv2 */
-			if ((SSL_OP_NO_SSLv2 & SSL_CTX_set_options(s->ssl_ctx, SSL_OP_NO_SSLv2)) != SSL_OP_NO_SSLv2) {
-				log_error_write(srv, __FILE__, __LINE__, "ss", "SSL:",
-						ERR_error_string(ERR_get_error(), NULL));
-				return -1;
-			}
-		}
-
-		if (!s->ssl_use_sslv3 && 0 != SSL_OP_NO_SSLv3) {
-			/* disable SSLv3 */
-			if ((SSL_OP_NO_SSLv3 & SSL_CTX_set_options(s->ssl_ctx, SSL_OP_NO_SSLv3)) != SSL_OP_NO_SSLv3) {
-				log_error_write(srv, __FILE__, __LINE__, "ss", "SSL:",
-						ERR_error_string(ERR_get_error(), NULL));
-				return -1;
-			}
-		}
+//		if (!s->ssl_use_sslv2 && 0 != SSL_OP_NO_SSLv2) {
+//			/* disable SSLv2 */
+//			if ((SSL_OP_NO_SSLv2 & SSL_CTX_set_options(s->ssl_ctx, SSL_OP_NO_SSLv2)) != SSL_OP_NO_SSLv2) {
+//				log_error_write(srv, __FILE__, __LINE__, "ss", "SSL:",
+//						ERR_error_string(ERR_get_error(), NULL));
+//				return -1;
+//			}
+//		}
+//
+//		if (!s->ssl_use_sslv3 && 0 != SSL_OP_NO_SSLv3) {
+//			/* disable SSLv3 */
+//			if ((SSL_OP_NO_SSLv3 & SSL_CTX_set_options(s->ssl_ctx, SSL_OP_NO_SSLv3)) != SSL_OP_NO_SSLv3) {
+//				log_error_write(srv, __FILE__, __LINE__, "ss", "SSL:",
+//						ERR_error_string(ERR_get_error(), NULL));
+//				return -1;
+//			}
+//		}
 
 		if (!buffer_string_is_empty(s->ssl_cipher_list)) {
 			/* Disable support for low encryption ciphers */
