diff -Naur ircd-hybrid-7.3.1.orig/src/rsa.c ircd-hybrid-7.3.1/src/rsa.c
--- ircd-hybrid-7.3.1.orig/src/rsa.c	2011-08-18 16:21:47.000000000 +0000
+++ ircd-hybrid-7.3.1/src/rsa.c	2016-12-15 06:30:11.353521497 +0000
@@ -166,13 +166,13 @@
 int
 get_randomness(unsigned char *buf, int length)
 {
-    /* Seed OpenSSL PRNG with EGD enthropy pool -kre */
-    if (ConfigFileEntry.use_egd &&
-        (ConfigFileEntry.egdpool_path != NULL))
-    {
-      if (RAND_egd(ConfigFileEntry.egdpool_path) == -1)
-            return -1;
-    }
+//    /* Seed OpenSSL PRNG with EGD enthropy pool -kre */
+//    if (ConfigFileEntry.use_egd &&
+//        (ConfigFileEntry.egdpool_path != NULL))
+//    {
+//      if (RAND_egd(ConfigFileEntry.egdpool_path) == -1)
+//            return -1;
+//    }
 
   if (RAND_status())
     return (RAND_bytes(buf, length));
