diff -Naur ikiwiki-3.20170111.orig/IkiWiki/Setup/Automator.pm ikiwiki-3.20170111/IkiWiki/Setup/Automator.pm
--- ikiwiki-3.20170111.orig/IkiWiki/Setup/Automator.pm	2017-01-11 18:16:53.000000000 +0000
+++ ikiwiki-3.20170111/IkiWiki/Setup/Automator.pm	2017-01-27 19:38:53.000000000 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/mnt/livecd/usr/bin/perl
 # Ikiwiki setup automator.
 
 package IkiWiki::Setup::Automator;
@@ -156,26 +156,26 @@
 		
 		if (! defined IkiWiki::emailuser($admin)) {
 			# Prompt for password w/o echo.
-			my ($password, $password2);
-			system('stty -echo 2>/dev/null');
-			local $|=1;
-			print "\n\nCreating wiki admin $admin ...\n";
-			for (;;) {
-				print "Choose a password: ";
-				chomp($password=<STDIN>);
-				print "\n";
-				print "Confirm password: ";
-				chomp($password2=<STDIN>);
-	
-				last if $password2 eq $password;
-	
-				print "Password mismatch.\n\n";
-			}
-			print "\n\n\n";
-			system('stty sane 2>/dev/null');
+#			my ($password, $password2);
+#			system('stty -echo 2>/dev/null');
+#			local $|=1;
+#			print "\n\nCreating wiki admin $admin ...\n";
+#			for (;;) {
+#				print "Choose a password: ";
+#				chomp($password=<STDIN>);
+#				print "\n";
+#				print "Confirm password: ";
+#				chomp($password2=<STDIN>);
+#	
+#				last if $password2 eq $password;
+#	
+#				print "Password mismatch.\n\n";
+#			}
+#			print "\n\n\n";
+#			system('stty sane 2>/dev/null');
 
 			if (IkiWiki::userinfo_setall($admin, { regdate => time }) &&
-			    IkiWiki::Plugin::passwordauth::setpassword($admin, $password)) {
+			    IkiWiki::Plugin::passwordauth::setpassword($admin, $config{adminpw})) {
 				IkiWiki::userinfo_set($admin, "email", $config{adminemail}) if defined $config{adminemail};
 			}
 			else {
