diff -Naur ikiwiki-3.20170111.orig/IkiWiki.pm ikiwiki-3.20170111/IkiWiki.pm
--- ikiwiki-3.20170111.orig/IkiWiki.pm	2017-01-11 18:16:53.000000000 +0000
+++ ikiwiki-3.20170111/IkiWiki.pm	2017-01-28 11:26:45.521116104 +0000
@@ -1405,9 +1405,9 @@
 						do => "create",
 						page => $link,
 						from => $lpage
-					)."\" rel=\"nofollow\">?</a>";
+					)."\" rel=\"nofollow\">$linktext</a>";
 			}
-			return "<span class=\"createlink\">$cgilink$linktext</span>"
+			return "$cgilink"
 		}
 	}
 	
