diff -Naur gtk+-3.22.26.orig/gtk/a11y/gtkaccessibility.c gtk+-3.22.26/gtk/a11y/gtkaccessibility.c
--- gtk+-3.22.26.orig/gtk/a11y/gtkaccessibility.c	2017-10-02 16:00:49.000000000 +0000
+++ gtk+-3.22.26/gtk/a11y/gtkaccessibility.c	2018-04-27 09:13:41.540985009 +0000
@@ -37,9 +37,9 @@
 #include <gtk/gtktogglebutton.h>
 #include <gtk/gtkaccessible.h>
 
-#ifdef GDK_WINDOWING_X11
-#include <atk-bridge.h>
-#endif
+//#ifdef GDK_WINDOWING_X11
+//#include <atk-bridge.h>
+//#endif
 
 static gboolean gail_focus_watcher      (GSignalInvocationHint *ihint,
                                          guint                  n_param_values,
@@ -988,9 +988,9 @@
   _gtk_accessibility_override_atk_util ();
   do_window_event_initialization ();
 
-#ifdef GDK_WINDOWING_X11
-  atk_bridge_adaptor_init (NULL, NULL);
-#endif
+//#ifdef GDK_WINDOWING_X11
+//  atk_bridge_adaptor_init (NULL, NULL);
+//#endif
 
   atk_misc_instance = g_object_new (GTK_TYPE_MISC_IMPL, NULL);
 }
diff -Naur gtk+-3.22.26.orig/configure gtk+-3.22.26/configure
--- gtk+-3.22.26.orig/configure	2018-04-27 09:15:02.533984807 +0000
+++ gtk+-3.22.26/configure	2018-04-27 09:31:55.921982280 +0000
@@ -27006,7 +27006,7 @@
 ########################################
 
 if test x$enable_x11_backend = xyes; then
-   ATK_PACKAGES="atk atk-bridge-2.0"
+   ATK_PACKAGES="atk"
 else
    ATK_PACKAGES="atk"
 fi
