diff -Naur mutt-1.6.1.orig/send.c mutt-1.6.1/send.c
--- mutt-1.6.1.orig/send.c	2016-04-30 21:07:24.000000000 +0000
+++ mutt-1.6.1/send.c	2017-01-31 03:55:04.379697008 +0000
@@ -1495,7 +1495,7 @@
    * This is done after allowing the user to edit the message so that security
    * settings can be configured with send2-hook and $edit_headers.
    */
-  if (WithCrypto && (msg->security == 0) && !(flags & (SENDBATCH | SENDMAILX | SENDPOSTPONED | SENDRESEND)))
+  if (WithCrypto && (msg->security == 0) && !(flags & (SENDMAILX | SENDPOSTPONED | SENDRESEND)))
   {
     if (option (OPTCRYPTAUTOSIGN))
       msg->security |= SIGN;
